### PR TITLE
Add tests for kv

### PR DESCRIPTION
**Pull request by Nek-**
This is #4 with:
- Rebase
- Consul binary
- Tests green
- Travis configuration

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/.composer.lock
+.composer.lock
```
Comment (lyrixx) [inline .gitignore]: Why ?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+php:
+    - 7.1
+    - 7.2
+
+branches:
+    only:
+        - master
+
+install:
+    - composer install --prefer-source
```
Comment (lyrixx) [inline .travis.yml]: why prefer-source ?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+script:
+    - ./vendor/bin/phpunit -c phpunit.xml.dist
```
Comment (lyrixx) [inline .travis.yml]: ` -c phpunit.xml.dist` is not needed

File: Consul/Tests/Services/AbstractTest.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+abstract class AbstractTest extends \PHPUnit_Framework_TestCase
+{
+    private static $consul;
+
+    public static function setUpBeforeClass()
+    {
+        static::clean(true);
+        $os = PHP_OS === 'Darwin' ? 'mac' : 'linux';
+        self::$consul = new Process('exec ' . __DIR__ . '/../../../bin/consul/consul_' . $os . ' agent -dev -bind=127.0.0.1');
```
Comment (lyrixx) [inline Consul/Tests/Services/AbstractTest.php]: I would use double quote instead of concatenation

File: Consul/Tests/Services/AbstractTest.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        self::$consul->start();
+
+        // This would be really better with https://github.com/symfony/symfony/pull/27742
+        $wait = true;
+        do {
+            self::$consul->checkTimeout();
+            $output = self::$consul->getOutput();
+
+            if (StringTools::contains($output, 'Synced node info')) {
```
Comment (lyrixx) [inline Consul/Tests/Services/AbstractTest.php]: I'm :-1: for adding a dep for only replacing a strpos.

File: Consul/Tests/Services/AbstractTest.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    public static function setUpBeforeClass()
+    {
+        static::clean(true);
+        $os = PHP_OS === 'Darwin' ? 'mac' : 'linux';
+        self::$consul = new Process('exec ' . __DIR__ . '/../../../bin/consul/consul_' . $os . ' agent -dev -bind=127.0.0.1');
+        self::$consul->start();
+
+        // This would be really better with https://github.com/symfony/symfony/pull/27742
+        $wait = true;
```
Comment (lyrixx) [inline Consul/Tests/Services/AbstractTest.php]: no need for `$wait`. Use `while (true)`, and `break` when it's ready

File: .gitignore (hunk 1 above)
Comment (Nek-) [reply .gitignore]: Well, this is wrong, and still wrong. I need to fix it. (the `.` is wrong)

File: .travis.yml (hunk 2 above)
Comment (Nek-) [reply .travis.yml]: Copy/paste 👼 I'll change to prefer-dist which is better here ofc.

File: .travis.yml (hunk 3 above)
Comment (Nek-) [reply .travis.yml]: well, also not too much

File: Consul/Tests/Services/AbstractTest.php (hunk 5 above)
Comment (Nek-) [reply Consul/Tests/Services/AbstractTest.php]: It's actually more: https://github.com/Nekland/Tools/blob/master/src/StringTools.php#L80 . And the dep is only for dev.

But I can remove it, simple strpos should be ok here.

File: Consul/Tests/Services/AbstractTest.php (hunk 6 above)
Comment (Nek-) [reply Consul/Tests/Services/AbstractTest.php]: 👍

**Comment by Nek-** [comment]
@lyrixx fixed

**Comment by lyrixx** [comment]
Thanks Maxime for working on this feature, this is much appreciated.

**Comment by lyrixx** [comment]
@fabpot Could you enable travis for this repository ? Thanks

**Pull request merged**